Formula/samba.rb

```diff
@@ -0,0 +1,54 @@
+class Samba < Formula
+  desc "SMB/CIFS file, print, and login server for UNIX"
+  homepage "https://samba.org/"
+  license "GPL-3.0-or-later"
+  url "https://download.samba.org/pub/samba/samba-4.14.5.tar.gz"
+  sha256 "bb6ef5d2f16b85288d823578abc453d9a80514c42e5a2ea2c4e3c60dc42335c3"
+
+  depends_on "pkg-config" => :build
+  depends_on "gnutls"
+  depends_on "openssl"
+  depends_on "readline" # Without the readline dependency the build fails on macOS 10.14+
+
+  # https://bugzilla.samba.org/show_bug.cgi?id=14680
+  patch do
+    url "https://attachments.samba.org/attachment.cgi?id=16579"
+    sha256 "86fce5306349d1c8f3732ca978a31065df643c8770114dc9d068b7b4dfa7d282"
+  end
+
+  def install
+    # Samba can be used to share directories with the guest in QEMU user-mode (SLIRP) networking
+    # with the `-net nic -net user,smb=/share/this/with/guest` option.
+    # The shared folder appears in the guest as "\\10.0.2.4\qemu".
+    system "./configure",
+           "--disable-cephfs",
+           "--disable-cups",
+           "--disable-iprint",
+           "--disable-glusterfs",
+           "--disable-python",
+           "--without-acl-support",
+           "--without-ad-dc",
+           "--without-ads",
+           "--without-dnsupdate",
+           "--without-ldap",
+           "--without-libarchive",
+           "--without-json",
+           "--without-ntvfs-fileserver",
+           "--without-pam",
+           "--without-regedit",
+           "--without-syslog",
+           "--without-utmp",
+           "--without-winbind",
+           "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+    # macOS has its own SMB daemon as /usr/sbin/smbd, so rename our smbd to samba-dot-org-smbd to avoid conflict.
+    # samba-dot-org-smbd is used by qemu.rb .
+    mv "#{sbin}/smbd", "#{sbin}/samba-dot-org-smbd"
```

Comment:
This means the formulae should be keg-only instead of renaming IYAM

Reply:
QEMU requires the binary to be renamed: https://github.com/Homebrew/homebrew-core/blob/0b1849e5735f6b1e59f5478832107c45b52e9e61/Formula/qemu.rb#L67-L72

Reply:
I think it should also be fine if it's called smbd but is inside a separate directory (i.e. not linked to /usr/local/sbin). I believe the reason I used a separate name was to make the error message more obvious.

Reply:
> QEMU requires the binary to be renamed

I think if we ship samba again we can just update the qemu formula

Reply:
> I think if we ship samba again we can just update the qemu formula

Let's not do this. This breaks the install of anyone using a non-Homebrew Samba with Homebrew QEMU. Personally I think renaming one file to avoid a conflict is perfectly fine. We have a lot of formulae that do more than this to avoid becoming keg-only too.

Reply:
Maybe we can symlink it?

Reply:
macOS version of `smbd` is completely different from Samba version of `smbd` except the binary name.

So I think renaming the binary without making it keg-only is more preferable than symlinking keg-only binary.